presto-spi/src/main/java/com/facebook/presto/spi/statistics/ColumnStatistics.java

```diff
@@ -14,30 +14,38 @@
 
 package com.facebook.presto.spi.statistics;
 
+import java.util.HashMap;
+import java.util.Map;
+
 import static com.facebook.presto.spi.statistics.Estimate.unknownValue;
+import static java.util.Collections.unmodifiableMap;
 import static java.util.Objects.requireNonNull;
 
 public final class ColumnStatistics
 {
-    public static final ColumnStatistics EMPTY_STATISTICS = ColumnStatistics.builder().build();
-
-    private final Estimate dataSize;
-    private final Estimate nullsCount;
-    private final Estimate distinctValuesCount;
+    private final Map<String, Estimate> statistics;
```

Comment:
I do not like the String keyed maps as public class interface. 
If you really want to have map here go for enums. Though I would opt for getters. 
IMO there is no benefit from map in typical use cases of `ColumnStatistics` class. And getters play much better with IDE completion.

You can use reflection based listing of stats if you really need that here. But hardcoding is also fine for me.